registry/repository.go

```diff
@@ -16,15 +16,31 @@ const (
 	maxManifestSizeLimit = 4 * 1024 * 1024  // 4 MiB
 )
 
+// RepositoryOptions provides user options when creating a Repository
+type RepositoryOptions struct {
+	// OCIimageManifest specifies if user wants to use OCI image manifest
+	// to store signatures in remote registries.
+	// By default, Notation will use OCI artifact manifest to store signatures.
+	// If OCIimageManifest flag is set to true, Notation will instead use
+	// OCI image manifest.
+	// Note, Notation will not automatically convert between these two types
+	// on any occasion.
+	// OCI artifact manifest: https://github.com/opencontainers/image-spec/blob/v1.1.0-rc2/artifact.md
+	// OCI image manifest: https://github.com/opencontainers/image-spec/blob/v1.1.0-rc2/manifest.md
+	OCIimageManifest bool
+}
+
 // repositoryClient implements Repository
 type repositoryClient struct {
 	registry.Repository
+	RepositoryOptions
 }
 
-// NewRepository returns a new Repository
-func NewRepository(repo registry.Repository) Repository {
+// NewRepositoryWithOptions returns a new Repository.
+func NewRepositoryWithOptions(repo registry.Repository, opts RepositoryOptions) Repository {
```

Comment:
We may still want to keep the original constructor so that we will have 2 constructors.

Reply:
In that case, do we still need the `RepositoryOptions`? Looks like `OCIImageManifest == false` has the same behavior as the original constructor. So an alternative might be:
```
// repositoryClient implements Repository
type repositoryClient struct {
	registry.Repository
	OCIImageManifest bool
}

func NewRepository(repo registry.Repository) Repository {
	return &repositoryClient{
		Repository: repo,
	}
}

func NewRepositoryWithImageManifest(repo registry.Repository) Repository {
	return &repositoryClient{
		Repository:       repo,
		OCIImageManifest: true,
	}
}
```

Reply:
The constructor
```go
func NewRepository(repo registry.Repository) Repository
```
should be used for most users with all default settings.

The constructor
```go
func NewRepositoryWithOptions(repo registry.Repository, opts RepositoryOptions) Repository
```
should be used for advanced users. Since we may introduce more options in the future, `RepositoryOptions` is designed for extensibility.

Reply:
I see. Will update the code.